### PR TITLE
Better handling of accidental IC in OOC 

### PR DESCRIPTION
**Pull request by wilfredrickson**
### Intent of your Pull Request

Created a Regular Expression to handle accidental In Character chat in Out Of Character chat, while fixing false positives that existed in the previous implementation.
Tested and working.

#### Changelog

:cl:  
bugfix: The OOC Ick-ock detector should be a bit smarter, and should stop asking you if you meant to say ":)" in IC.
/:cl:

### CODE REVIEW

**Comment by Altoids1** [comment]
image

File: code/modules/client/verbs/ooc.dm [hunk 1]
```diff
@@ -34,7 +34,9 @@
 	msg = pretty_filter(msg) //yogs
 	msg = emoji_parse(msg)
 
-	if((copytext(msg, 1, 2) in list(".",";",":","#")) || (findtext(lowertext(copytext(msg, 1, 5)), "say")))
+	var/regex/ickock = regex(@"^\s*(#.*|,.*|(:|;)(\w|\s|\d)|(say \x22)|\.\.?(?!\.))","i")
+	//captures a lot of accidental in character speech in ooc chat
+	if(ickock.Find(msg))
```
Comment (Altoids1) [inline code/modules/client/verbs/ooc.dm]:
```suggestion
	if(length(msg) > 4 && ickock.Find(msg))
```
Just to make this a tad bit more robust against emoticon usage.

Also, add some ``//yogs start`` and ``//yogs end`` around this change.

File: code/modules/client/verbs/ooc.dm [hunk 2]
```diff
@@ -80,7 +82,7 @@
 		else // Else if they're an AdminObserver
 			oocmsg += "<span class='adminobserverooc'><span class='prefix'>[find_admin_rank(src)]" // The header for an AO's OOC.
 		//Check yogstation\code\module\client\verbs\ooc for the find_admin_rank definition.
-
```
Comment (Altoids1) [inline code/modules/client/verbs/ooc.dm]: Remove this whitespace change.

File: code/modules/client/verbs/ooc.dm [hunk 3]
```diff
@@ -96,7 +98,7 @@
 			oocmsg = "<font color='[GLOB.normal_ooc_colour]'>"
 		oocmsg += "<span class='ooc'>[is_donator(src) ? "(Donator)" : ""]<span class='prefix'>OOC:</span> <EM>[keyname]:</EM> <span class='message linkify'>[msg]</span></span></font>"
 		oocmsg_toadmins = oocmsg
-
```
Comment (Altoids1) [inline code/modules/client/verbs/ooc.dm]: Remove this whitespace change.

**Review by Altoids1 — CHANGES_REQUESTED** [review]

**Comment by wilfredrickson** [comment]
Hopefully it's all in working order now, sorry for the fuss

**Review by Altoids1 — APPROVED** [review]

**Pull request merged**